src/sentry/utils/request_cache.py

```diff
@@ -0,0 +1,33 @@
+import threading
+
+from django.core.signals import request_finished
+
+from sentry.models import OrganizationMember
+
+_cache = threading.local()
+
+
+def request_cache(func):
```

Comment:
Just a reminder from slack:
We should do a `if sentry.app.env.request is None` check and totally skip the cache if there's no request.